roles/bind-unifiedpush-apb/tasks/main.yml

```diff
@@ -14,8 +14,27 @@
     dest: /tmp/secret.yaml
   when: passphrase is defined
 
-- name: "Create binding secret"
+- name: "Get the secrets that currently exist"
+  shell: "oc get secrets --selector='clientId={{ clientId }},serviceName=ups' -o jsonpath='{.items[0].data.config}'  -n {{ namespace }} | base64 --decode "
+  register: config
+  ignore_errors: yes
+
+- name: "Create empty json object to avoid failing on first binding"
+  set_fact:
+    config_json: "{}|from_json"
+
+- name: "Convert config to json format"
+  set_fact:
+    config_json: "{{ config.stdout|from_json }}"
+  ignore_errors: yes
+
+- name: "Create Android binding secret"
+  shell: "oc create -f /tmp/secret.yaml -n {{ namespace }}"
+  when: config_json.android is undefined and googlekey is defined
+
+- name: "Create iOS binding secret"
   shell: "oc create -f /tmp/secret.yaml -n {{ namespace }}"
+  when: config_json.ios is undefined and passphrase is defined
```

Comment:
Might be a silly question but why are we checking `config_json.ios is undefined` when we want to create an iOS secret? I see the same thing for the android one.

These checks are confusing to me. It might be better to check value of the `CLIENT_TYPE` parameter. e.g. `CLIENT_TYPE is Android` or `CLIENT_TYPE is iOS`

Reply:
This is done because we need the secret to get created on the first pass. after this - the secret will have ios defined so this will never happen again. This ensures that the secret is not created again (which was overriding the existing one). We cannot block the user from trying to create a second binding on the Openshift UI - so we need to ensure it does not override the existing ios config in the secret

Reply:
@darahayes  I'm open to suggestions of a better way of doing this.